BUG: "No module named cython" when building 26.0.0+ with Pyston 2.3.5

### This is a pyzmq bug

- [X] This is a pyzmq-specific bug, not an issue of zmq socket behavior. Don't worry if you're not sure! We'll figure it out together.

### What pyzmq version?

26.0.0+

### What libzmq version?

4.3.4

### Python version (and how it was installed)

pyston-2.3.5 installed via pyenv

### OS

Ubuntu 22.04

### What happened?

When updating pyzmq from version 25.1.2 I got an error.

### Code to reproduce bug

```python
1. Install pyston-2.3.5 from pyenv
2. Make a virtualenv
3. Try to install pyzmq
```


### Traceback, if applicable

```shell
(more can be provided on request)
...
  [103/109] Building CXX object _deps/bundled_libzmq-build/CMakeFiles/objects.dir/src/tipc_connecter.cpp.o
  [104/109] Building CXX object _deps/bundled_libzmq-build/CMakeFiles/objects.dir/src/udp_engine.cpp.o
  [105/109] Building CXX object _deps/bundled_libzmq-build/CMakeFiles/objects.dir/src/socket_base.cpp.o
  [106/109] Linking CXX static library _deps/bundled_libzmq-build/lib/libzmq.a
  [107/109] Generating _src/_zmq.c
  FAILED: _src/_zmq.c /tmp/tmpd9xa1398/build/_src/_zmq.c
  cd /tmp/tmpd9xa1398/build && /home/davidlaw/.pyenv/versions/pyston-2.3.5/envs/pyzmq-test-pyston-2.3.5/bin/python -mcython --3str --output-file /tmp/tmpd9xa1398/build/_src/_zmq.c --module-name zmq.backend.cython._zmq /tmp/pip-install-m1z8ag2s/pyzmq_22e735db494c49e7b0d28815b1ad745c/zmq/backend/cython/_zmq.py
  /home/davidlaw/.pyenv/versions/pyston-2.3.5/envs/pyzmq-test-pyston-2.3.5/bin/python: No module named cython
  ninja: build stopped: subcommand failed.

  *** CMake build failed
  error: subprocess-exited-with-error
  
  × Building wheel for pyzmq (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /home/davidlaw/.pyenv/versions/pyston-2.3.5/envs/pyzmq-test-pyston-2.3.5/bin/python /home/davidlaw/.pyenv/versions/pyston-2.3.5/envs/pyzmq-test-pyston-2.3.5/lib/python3.8-pyston2.3/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py build_wheel /tmp/tmpsj5fhcnz
  cwd: /tmp/pip-install-m1z8ag2s/pyzmq_22e735db494c49e7b0d28815b1ad745c
  Building wheel for pyzmq (pyproject.toml) ... error
  ERROR: Failed building wheel for pyzmq
Failed to build pyzmq
ERROR: Could not build wheels for pyzmq, which is required to install pyproject.toml-based projects
Exception information:
Traceback (most recent call last):
  File "/home/davidlaw/.pyenv/versions/pyston-2.3.5/envs/pyzmq-test-pyston-2.3.5/lib/python3.8-pyston2.3/site-packages/pip/_internal/cli/base_command.py", line 180, in exc_logging_wrapper
    status = run_func(*args)
  File "/home/davidlaw/.pyenv/versions/pyston-2.3.5/envs/pyzmq-test-pyston-2.3.5/lib/python3.8-pyston2.3/site-packages/pip/_internal/cli/req_command.py", line 245, in wrapper
    return func(self, options, args)
  File "/home/davidlaw/.pyenv/versions/pyston-2.3.5/envs/pyzmq-test-pyston-2.3.5/lib/python3.8-pyston2.3/site-packages/pip/_internal/commands/install.py", line 429, in run
    raise InstallationError(
pip._internal.exceptions.InstallationError: Could not build wheels for pyzmq, which is required to install pyproject.toml-based projects
Remote version of pip: 24.0
Local version of pip:  24.0
Was pip installed by pip? True
Removed build tracker: '/tmp/pip-build-tracker-ih8jssr0'
```
```


### More info

I have tried installing in a clean virtualenv, and again after manually installing Cython. I got the same result both times.

25.1.2 installs fine.

--- a/docs/source/changelog.md
+++ b/docs/source/changelog.md
@@ -7,6 +7,10 @@ For a full changelog, consult the [git log](https://github.com/zeromq/pyzmq/comm
 
 ## 26
 
+### 26.0.3
+
+- Add Cython as build dependency for non-pypy, non-cpython implementations (fixes install on pyston)
+
 ### 26.0.2
 
 - When bundling libsodium, download from libsodium's releases on GitHub instead of download.libsodium.org,

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,10 +2,8 @@
 [build-system]
 requires = [
   "cffi; implementation_name == 'pypy'",
-  "cython>=3.0.0; implementation_name == 'cpython'",
+  "cython>=3.0.0; implementation_name != 'pypy'",
   "packaging",
-  # "setuptools>=61",
-  # "setuptools_scm[toml]",
   "scikit-build-core",
 ]
 build-backend = "scikit_build_core.build"

